File: InternetBulletin.Shared/DTOs/Posts/PostRatingDTO.cs
// *********************************************************************************
//	<copyright file="PostRatingDTO.cs" company="Personal">
//		Copyright (c) 2025 Personal
//	</copyright>
// <summary>The Post Ratings DTO.</summary>
// *********************************************************************************

namespace InternetBulletin.Shared.DTOs.Posts
{
    /// <summary>
    /// The Post Ratings DTO.
    /// </summary>
    public class PostRatingDTO
    {
        /// <summary>
        /// Gets or sets the post id.
        /// </summary>
        /// <value>
        /// The post id.
        /// </value>
        public string PostId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the is increment.
        /// </summary>
        /// <value>
        /// The incremented or decremented value.
        /// </value>
        public bool IsIncrement { get; set; }
    }
}

File: IBBS.UI/Models/PostRatingDto.ts
/**
 * PostRatingDtoModel class to represent the post rating dto.
 */
class PostRatingDtoModel {
    /**
     * Gets or sets the post id.
     */
    postId: string = "";

    /**
     * Gets or sets the incremented or decremented value
     */
    isIncrement: boolean = false;

    /**
     * Creates an instance of PostRatingDtoModel
     * @param {string} PostId The post id.
     * @param {boolean} IsIncrement The incremented or decremented value.
     */
    constructor(PostId: string, IsIncrement: boolean) {
        this.postId = PostId;
        this.isIncrement = IsIncrement;
    }
}

export default PostRatingDtoModel;
